pkg/patch/patch.go

```diff
@@ -31,28 +34,31 @@ func GeneratePatch(meta Meta) ([]byte, error) {
 		initPatch = fmt.Sprintf(create, "initContainers", initPatch)
 	}
 
-	sidecarPatch, err := renderSidecar(meta.Sidecar)
+	sidecarPatches, err := renderSidecars(meta.Sidecar)
 	if err != nil {
 		return []byte(patch), err
 	}
 
-	if meta.AppendSidecar {
-		sidecarPatch = fmt.Sprintf(add, "containers", sidecarPatch)
-	} else {
-		sidecarPatch = fmt.Sprintf(create, "containers", sidecarPatch)
+	for i := range sidecarPatches {
+		if meta.AppendSidecar {
+			sidecarPatches[i] = fmt.Sprintf(add, "containers", sidecarPatches[i])
+		} else {
+			sidecarPatches[i] = fmt.Sprintf(create, "containers", sidecarPatches[i])
+		}
```

Comment:
This logic might need to be tweaked. 

If the user has not containers initially, `meta.AppendSidecar` will be false, and we will end up **creating** two sidecars, instead of creating one and appending others. 

Do you mind adding a test to either catching the bug or proving I was wrong?

Reply:
Yes, I have my doubts here on how this works. Will need to do end-to-end testing before I can finalize this PR.

Reply:
The new change looks nice!

For those who are curious, [jsonpatch.com](http://jsonpatch.com/) is a good site explaining how json patch works. 

```
Add
{ "op": "add", "path": "/biscuits/1", "value": { "name": "Ginger Nut" } }
Adds a value to an object or inserts it into an array. In the case of an array, the value is inserted before the given index. The - character can be used instead of an index to insert at the end of an array.
```